packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
+      }
+    } else if (dateFormatOrder === 'DMY') {
+      if ((event.keyCode === KeyCode.KEY_LEFT || event.keyCode === KeyCode.KEY_DELETE || event.keyCode === KeyCode.KEY_BACK_SPACE) && yearInputRef.current.selectionEnd === 0) {
+        setInputFocus(event, monthInputRef.current, date.month.length, date.month.length);
+      }
+    } else if (dateFormatOrder === 'YMD') {
+      if (event.keyCode === KeyCode.KEY_RIGHT && yearInputRef.current.selectionEnd === date.year.length) {
+        setInputFocus(event, monthInputRef.current, 0, 0);
+      }
+    }
+  };
+
+  const handleOnInputFocus = (event, type) => {
+    if (onFocus) {
+      onFocus(event);
+    }
+
+    setFocused(true);
+
+    if (type === DateUtil.inputType.DAY) {
+      setDayInitialFocused(true);
+      dayInputRef.current.setSelectionRange(0, dayInputRef.current.value.length);
+    } else if (type === DateUtil.inputType.MONTH) {
+      setMonthInitialFocused(true);
+      monthInputRef.current.setSelectionRange(0, monthInputRef.current.value.length);
+    } else {
+      setYearInitialFocused(true);
+      yearInputRef.current.setSelectionRange(0, yearInputRef.current.value.length);
+    }
+  };
+
+  const handleOnInputBlur = (event, type) => {
+    if (onBlur) {
+      onBlur(event);
+    }
+
+    setFocused(false);
+
+    if (type === DateUtil.inputType.DAY) {
+      setDayInitialFocused(false);
+    } else if (type === DateUtil.inputType.MONTH) {
+      setMonthInitialFocused(false);
+    } else {
+      setYearInitialFocused(false);
+    }
+
+    if (type === DateUtil.inputType.DAY || type === DateUtil.inputType.MONTH) {
+      let inputValue = event.target.value;
+
+      // Prepend a 0 to the value when losing focus and the value is single digit.
+      if (inputValue.length === 1) {
+        inputValue = inputValue === '0' ? '' : '0'.concat(inputValue);
+
+        handleDateChange(event, inputValue, type);
+      }
+    }
+  };
+
+  const handleOnButtonClick = (event) => {
+    const attributes = { ...inputAttributes };
+    if (!attributes.readOnly && onCalendarButtonClick && onClick) {
+      onCalendarButtonClick(event, onClick);
+    }
+  };
+
+  const handleOnButtonKeyDown = (event) => {
+    if (onKeyDown) {
+      onKeyDown(event);
+    }
+  };
+
+  const dateInputContainerClasses = cx([
+    'date-input-container',
+    theme.className,
+  ]);
+
+  const dateInputClasses = cx([
+    'date-input',
+    { 'is-focused': isFocused },
+    { 'is-invalid': isInvalid },
+    { 'is-incomplete': isIncomplete && required && !isInvalid },
+  ]);
+
+  const buttonClasses = cx([
+    'button',
+    { 'is-invalid': isInvalid },
+  ]);
+
+  const dayInputClasses = cx([
+    'date-input-day',
+    { 'initial-focus': dayInitialFocused },
+  ]);
+
+  const monthInputClasses = cx([
+    'date-input-month',
+    { 'initial-focus': monthInitialFocused },
+  ]);
+
+  const yearInputClasses = cx([
+    'date-input-year',
+    { 'initial-focus': yearInitialFocused },
+  ]);
```

Comment:
Can these share the same class of `date-input` for styles?

Reply:
Well most of the styles are the same, except for their respective widths. https://github.com/cerner/terra-framework/blob/ba9d3c9b977ded21c1db0789cbe2b1e43dd3c7c6/packages/terra-date-picker/src/DatePicker.module.scss#L128-L138